cpp/open3d/visualization/app/Viewer.cpp

```diff
@@ -0,0 +1,85 @@
+// ----------------------------------------------------------------------------
+// -                        Open3D: www.open3d.org                            -
+// ----------------------------------------------------------------------------
+// The MIT License (MIT)
+//
+// Copyright (c) 2018-2021 www.open3d.org
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+// ----------------------------------------------------------------------------
+
+#include "open3d/visualization/app/Viewer.h"
+
+#include <iostream>
+#include <string>
+
+#include "open3d/utility/Logging.h"
+#include "open3d/visualization/gui/Application.h"
+#include "open3d/visualization/gui/Native.h"
+#include "open3d/visualization/visualizer/GuiVisualizer.h"
+
+using namespace open3d;
+using namespace open3d::geometry;
+using namespace open3d::visualization;
+
+namespace {
+static const std::string gUsage = "Usage: Open3DViewer [meshfile|pointcloud]";
```

Comment:
1. Put inside `open3d::visualization::app`. I think it can be placed together with:
    ```cpp
    const int width = 1280;
    const int height = 960;
    ```
2. Rename `gUsage` to `snake_case`.